price-reporter/src/http_server/routes.rs

```diff
@@ -98,3 +103,83 @@ impl Handler for PriceHandler {
         }
     }
 }
+
+// -------------------------------
+// | REFRESH TOKEN MAPPING ROUTE |
+// -------------------------------
+
+/// The route for the token mapping refresh endpoint
+pub const REFRESH_TOKEN_MAPPING_ROUTE: &str = "/refresh-token-mapping";
+
+/// The handler for the token mapping refresh endpoint
+#[derive(Clone)]
+pub struct RefreshTokenMappingHandler {
+    /// The HMAC key for the admin API
+    admin_key: HmacKey,
+    /// The path to the token remap file
+    token_remap_path: Option<String>,
+    /// The chain to use for the token remap
+    remap_chain: Chain,
+    /// The global price streams
+    price_streams: GlobalPriceStreams,
+    /// The configuration for the exchange connections
+    config: ExchangeConnectionsConfig,
+}
+
+impl RefreshTokenMappingHandler {
+    /// Create a new token mapping refresh handler
+    pub fn new(
+        admin_key: HmacKey,
+        token_remap_path: Option<String>,
+        remap_chain: Chain,
+        price_streams: GlobalPriceStreams,
+        config: ExchangeConnectionsConfig,
+    ) -> Self {
+        Self { admin_key, token_remap_path, remap_chain, price_streams, config }
+    }
+
+    /// Authenticate a token mapping refresh request using the admin HMAC key
+    async fn authenticate_request(&self, req: &mut Request<Body>) -> Result<(), ServerError> {
+        let req_body = to_bytes(req.body_mut()).await.map_err(err_str!(ServerError::Serde))?;
+        let path = req.uri().path();
+        let headers = req.headers();
+        validate_expiring_auth(path, headers, &req_body, &self.admin_key)
+            .map_err(err_str!(ServerError::Unauthorized))
+    }
+
+    /// Refresh the token mapping from the remote source
+    pub async fn refresh_token_mapping(&self) -> Result<(), ServerError> {
+        let token_remap_path = self.token_remap_path.clone();
+        let remap_chain = self.remap_chain;
+        tokio::task::spawn_blocking(move || setup_token_remaps(token_remap_path, remap_chain))
+            .await
+            .map_err(err_str!(ServerError::TokenRemap))
+            .and_then(|res| res.map_err(err_str!(ServerError::TokenRemap)))?;
+
+        // Re-initialize the default price streams after refreshing the token mapping
+        init_default_price_streams(&self.price_streams, &self.config)
```

Comment:
qq for my own understanding: what will happen to the existing price streams? are they shut down?

Reply:
existing price streams are untouched, we only initialize new price streams if they are not already present in the global mapping